src/js/angular/utils/uri-utils.js

```diff
@@ -31,10 +31,17 @@ function UriUtils(ClassInstanceDetailsService) {
         return validProtocol && validPath;
     }
 
+    /**
+     * Validates if the given text is a valid IRI.
+     * @param {object} value
+     * @param {string} valueText
+     * @return {boolean} Returns true if the value is a valid IRI and false otherwise.
+     */
     function isValidIri(value, valueText) {
         if (value !== undefined && valueText !==undefined && (validateRdfUri(valueText) || !!valueText.match(iriRegExp))) {
             return true;
         }
+        return false;
```

Comment:
Should return true/false in order to be more predictable. Additionally, the `undefined` value returned by this function instead of `false` turned out to brake the validation directive which expects true/false